url969d1ddca4_compmstr_java_scheme_tgz-pJ8-com_undi_javascheme_JavaSchemeJ8/src/com/undi/util/Reflector.java

```diff
@@ -0,0 +1,371 @@
+      try{
+        return prepRet(f.getType(), f.get(target));
+      }catch(IllegalAccessException e){
+        throw SneakyThrow.sneakyThrow(e);
+      }
+    }
+    throw new IllegalArgumentException("No matching field found");
+  }
+  
+  public static Object setInstanceField(Object target, String fieldName, Object val){
+    Class<?> c = target.getClass();
+    Field f = getField(c, fieldName, false);
+    if(f != null){
+      try{
+        f.set(target, boxArg(f.getType(), val));
+      }catch(IllegalAccessException e){
+        throw SneakyThrow.sneakyThrow(e);
+      }
+      return val;
+    }
+    throw new IllegalArgumentException("No matching field found");
+  }
+  
+  public static Object invokeStaticMethodVariadic(String className, String methodName, Object... args){
+    return invokeStaticMethod(className, methodName, args);
+  }
+  
+  public static Object invokeStaticMethod(String className, String methodName, Object[] args){
+    try {
+      Class<?> c = Class.forName(className);
+      return invokeStaticMethod(c, methodName, args);
+    } catch (Exception e) {
+      throw SneakyThrow.sneakyThrow(e);
+    }
+  }
+  
+  public static Object invokeStaticMethod(Class<?> c, String methodName, Object[] args){
+    if(methodName.equals("new")){
+      return invokeConstructor(c, args);
+    }
+    List<Method> methods = getMethods(c, args.length, methodName, true);
+    return invokeMatchingMethod(methodName, methods, null, args);
+  }
+  
+  public static Object invokeConstructor(Class<?> c, Object[] args){
+    try{
+      Constructor<?>[] allctors = c.getConstructors();
+      ArrayList<Constructor<?>> ctors = new ArrayList<Constructor<?>>();
+      for(Constructor<?> ctor : allctors){
+        if(ctor.getParameterTypes().length == args.length){
+          ctors.add(ctor);
+        }
+      }
+      if(ctors.isEmpty()){
+        throw new IllegalArgumentException("Not Matching constructor found for: " + c);
+      }else if(ctors.size() == 1){
+        Constructor<?> ctor = (Constructor<?>) ctors.get(0);
+        return ctor.newInstance(boxArgs(ctor.getParameterTypes(), args));
+      }else{ //overloaded with same arity
+        for(Constructor<?> ctor : ctors){
+          Class<?>[] params = ctor.getParameterTypes();
+          if(isCongruent(params, args)){
+            Object[] boxedArgs = boxArgs(params, args);
+            return ctor.newInstance(boxedArgs);
+          }
+        }
+        throw new IllegalArgumentException("Not Matching constructor found for: " + c);
+      }
+    }catch(Exception e){
+      throw SneakyThrow.sneakyThrow(e);
+    }
+  }
+  
+  public static Object invokeInstanceMethod(Object target, String methodName, Object[] args){
+    Class<?> c = target.getClass();
+    List<Method> methods = getMethods(c, args.length, methodName, false);
+    return invokeMatchingMethod(methodName, methods, target, args);
+  }
+  
+  static public boolean paramArgTypeMatch(Class<?> paramType, Class<?> argType){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.